### PR TITLE
Added Crates

### LINKED ISSUE #41: Boxes
- Box implementation in game
- Box moveable/pickupable

### LINKED ISSUE #384: Boxes
As a player wearing the Oculus Rift, I want to have boxes added to the game, so that I can reach the treasure more easily.

This task has the following subtasks:
- #385 Boxes - Box entity
- #386 Boxes - Box spawn limitations

### LINKED ISSUE #385: Boxes - Box entity
Add a box entity to the game that can be picked up and thrown by the player, that will trigger certain other entities.

This is a subtask of #384 - Boxes

### LINKED ISSUE #386: Boxes - Box spawn limitations
Add a box placement cap to the elves.

This is a subtask of #384 - Boxes

### PR DESCRIPTION
**Pull request by ChielBruin**
Added crates to the game.
Crates can be picked up and thrown by the player.
When a crate lands on a pitfall or landmine, that entity will be triggered.

Closes #41, closes #384, closes #385 and closes #386.

### CODE REVIEW

File: src/main/java/nl/tudelft/contextproject/controller/GameController.java [hunk 1]
```diff
@@ -146,6 +147,10 @@ protected void attachLevel() {
 		AmbientLight al = new AmbientLight();
 		al.setColor(ColorRGBA.White.mult(.5f));
 		addLight(al);
+		
+		Entity e = new Crate();
```
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/controller/GameController.java]: Doesn't this mean there's always a create in the game?

File: src/main/java/nl/tudelft/contextproject/model/Inventory.java [hunk 2]
```diff
@@ -66,17 +65,27 @@ public void add(Bomb bomb) {
...
 			updateTickListeners();
 		}
 	}
 	
-
+	/**
+	 * Drop the holdable that the inventory holds.
+	 * 
+	 * @return
+	 * 		the holdable that was held or null when nothing was held.
```
Comment (Geweldig) [inline src/main/java/nl/tudelft/contextproject/model/Inventory.java]: "." should be removed

File: src/main/resources/webinterface/js/main.js [hunk 3]
```diff
@@ -544,9 +544,11 @@ function getClassForEntityType(entityType) {
             return "Invisible_Wall";
         case 12:
             return "Damaged_Wall";
+        case 13:
+            return "Crate";
         default:
             showAlert("Invalid tile type: " + entityType, "Danger");
-            throw "Invalid tile type: " + entityType;
+            //throw "Invalid tile type: " + entityType;
```
Comment (Geweldig) [inline src/main/resources/webinterface/js/main.js]: line can be removed

**Comment by Geweldig** [comment]
After my issues have been addressed I think this can be merged.

**Comment by Diocruel** [comment]
Looks great! Nice use of RigidBody constructor.

**Comment by Diocruel** [comment]
Good last fix btw, as I accidentally did not remove it during merging! Ready to merge after Travis.

**Comment by Diocruel** [comment]
I think it is ready to merge now that the build is fixed.

**Comment by Taeir** [comment]
Looking very good Chiel!

Merging this.

**Pull request merged**